### PR TITLE
Add a new web environment that runs without the WebExtension context

### LINKED ISSUE #411: Build a second bundle of NDT that is suitable for hosting on the web
NDT should have a second webpack bundle that is hostable as a web page. It should enable minification, and define a build-time constant that features can check to modify their behavior. It may need other differences as well.

### PR DESCRIPTION
**Pull request by mythmon**
Extension-only APIs have been shimmed to provide no results and print a
warning to the console. Future work should remove these parts of the site.

Fixes #411

### CODE REVIEW

**Comment by mythmon** [comment]
To test that this is really suitable to host as a static site, I copied the contents of the `dist-web` directroy into a Glitch app: https://purrfect-circular-astrodon.glitch.me

So far I've noticed that logging in doesn't work, but that's kind of expected since we were using web extensions APIs to do that.

File: webpack.config.js [hunk 1]
```diff
@@ -158,48 +131,168 @@ module.exports = async (env, argv = {}) => {
...
+                if (["addListener", "getRecipeSuitabilities"].includes(prop)) {
+                  return () => {};
+                }
+                return proxyMaker(prefix + prop + ".");
+              }
+            });
+          };
+          return proxyMaker("");
+        })())`.replace("\n", " "),
+      }),
```
Comment (rehandalal) [inline webpack.config.js:241]: 👏🏽

File: docs/dev.md [hunk 2]
```diff
@@ -1,84 +1,106 @@
...
+```sh
 $ git clone https://github.com/mozilla/normandy-devtools.git
 $ cd normandy-devtools
 $ yarn install
 ```
 
-To run the extension in development mode, you can simply use the `watch` command to
+## Extension
+
+To work on the extension version NDT, you'll need to have Firefox Nightly.
```
Comment (leplatrem) [inline docs/dev.md]: *of NDT* ?

File: docs/dev.md [hunk 3]
```diff
@@ -1,84 +1,106 @@
...
-```
 
-> NOTE: If you are seeing an error while trying to run `watch` make sure that Nightly
-> is installed on your system and if you are on Linux make sure that `firefox-nightly`
-> is on your PATH.
+> NOTE: If you are seeing an error while trying to run `watch-ext` make sure
+> that Nightly is installed on your system. If you are on Linux check that
+> `firefox-nightly` is on your PATH.
+
+## Web
```
Comment (leplatrem) [inline docs/dev.md:37]: *nipick* Above we said website, below webpage. They are obvious synonyms but I think wee could stick to one same word, no?

**Review by rehandalal — APPROVED** [review]
nice work. lgtm

bors d=@mythmon

**Comment by mythmon** [comment]
bors merge

**Comment by mythmon** [comment]
bors merge

**Pull request merged**